src/Traits/Livewire/CreatesDocuments.php

```diff
@@ -0,0 +1,234 @@
+                    array_keys($item->resolvePrintViews())
+                )
+            );
+
+            // create the documents
+            foreach ($createDocuments as $createDocument) {
+                $media = is_a($item, HasMedia::class) ? $item->getMedia($createDocument)->last() : null;
+                $isDownload = in_array($createDocument, data_get($this->selectedPrintLayouts, 'download', []));
+                $isPrint = in_array($createDocument, data_get($this->selectedPrintLayouts, 'print', []));
+                $isForce = in_array($createDocument, data_get($this->selectedPrintLayouts, 'force', []));
+                $isEmail = in_array($createDocument, data_get($this->selectedPrintLayouts, 'email', []));
+
+                if ((! $media && ($isDownload || $isPrint)) || $isForce || ! $async) {
+                    try {
+                        /** @var PrintableView $file */
+                        $file = Printing::make([
+                            'model_type' => $item->getMorphClass(),
+                            'model_id' => $item->getKey(),
+                            'view' => $createDocument,
+                        ])
+                            ->checkPermission()
+                            ->validate()
+                            ->execute();
+
+                        $media = $file->attachToModel($item);
+                    } catch (ValidationException|UnauthorizedException $e) {
+                        exception_to_notifications($e, $this);
+
+                        continue;
+                    }
+                }
+
+                if ($isDownload) {
+                    $downloadIds[] = $media->getKey();
+                }
+
+                if ($isPrint) {
+                    // TODO: add to print queue for spooler
+                    $printIds[] = $media->getKey();
+                }
+
+                if ($isEmail) {
+                    if ($media) {
+                        $mailAttachments[] = [
+                            'name' => $media->file_name,
+                            'id' => $media->getKey(),
+                        ];
+                    } else {
+                        $mailAttachments[] = $this->getCreateAttachmentArray($item, $createDocument);
+                    }
+                }
+            }
+
+            if (data_get($this->selectedPrintLayouts, 'email', false) && $mailAttachments) {
+                $item->refresh();
+                $bladeParameters = method_exists($this, 'getBladeParameters')
+                    ? $this->getBladeParameters($item)
+                    : [];
+
+                if ($bladeParameters instanceof SerializableClosure) {
+                    $bladeParameters = serialize($bladeParameters);
+                }
+
+                $mailAttachments[] = $this->getAttachments($item);
+                $mailMessages[] = [
+                    'to' => array_unique($this->getTo($item, $createDocuments)),
```

Comment:
I would put the responsibility on the implementation of the getTo() method and not do an extra array_unique().
If you start this, then you also have to do array_filter() and maybe other shenanigans to ensure that this array contains only valid values.

Reply:
√